sbt-ash/src/main/scala/org/virtuslab/ash/DumpEventSchema.scala

```diff
@@ -26,14 +26,12 @@ object DumpEventSchema {
       writer <-
         outputFile.createIfNotExists(createParents = true).clear().newFileOutputStream().printWriter().autoClosed
     } writer.print(yamlPrinter.pretty(json))
+    outputFile
   }
 
-  def dumpEventSchemaTask(key: TaskKey[Unit]): Def.Setting[_] =
+  def dumpEventSchemaTask(key: TaskKey[JFile]): Def.Setting[_] =
     key := {
-      val c = (Compile / compile).value
-      c.readCompilations()
-      DumpEventSchema(
-        (key / dumpEventSchemaOutputFile).value.toScala,
-        (key / dumpEventSchemaCompilerPluginOutputFile).value.toScala)
+      (Compile / compile).value
```

Comment:
Compiler warns about `a pure expression does nothing in statement position`... it's just being dumb, right? coz `.value` is not really a pure expression in this context right? 🤔

Reply:
Yeah, this is how you declare dependency in a key.